### PR TITLE
Line Chart and Area Chart - Component tests

### PR DESCRIPTION
**Pull request by v-baambati**

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/89f6b7c0835aca7ce29aff3c9a22ace4eb1b01d5'>89f6b7c0835aca7ce29aff3c9a22ace4eb1b01d5</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+**Line Chart – Component test plan**
+
+**Sub-components: Line, x-axis, y-axis, Event, Time Range, Callout, Drop down and Legend**
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:3]: what does time range and drop down mean. Is it the legend overflow dropdown.

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+|- With WrapXAxisLables set to “true”|Should wrap x axis label values|Enzyme|
+|- With yAxisTickFormat set to “%d”|<p>Should render the y-axis ticks in the format specified</p><p></p>|Enzyme|
+|Test 2: Basic props testing|||
+|- HideLegend prop set to “true”|Should not mount legend when hideLegend is true|Enzyme|
+|- HideLegend prop set to “false”|Should mount legend when hideLegend is false|Enzyme|
+|- HideTooltip prop set to “true”|Should not mount callout when hideTootip is true|Enzyme|
+|- HideTooltip prop set to “false”|Should mount callout when hideTootip is false|Enzyme|
+|- onRenderCalloutPerStack prop is not given|Should not render onRenderCalloutPerStack|Pending|
+|- onRenderCalloutPerDataPoint is given|Should render onRenderCalloutPerDataPoint|Pending|
+|- onRenderCalloutPerDataPoint is given|Should not render onRenderCalloutPerDataPoint|Pending|
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:28]: onRenderCalloutPerDataPoint is not given

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+|- No prop changes: Mount line chart and then set the same props again|Render function should have been called twice|Enzyme|
+|- Prop changes: Mount line chart and then set some other prop|Render function should have been called twice|Enzyme|
+|Test 4: Mouse events|||
+|- Mouse over on a line|Should render callout correctly on mouseover|Enzyme & RTL|
+|- Mouse move from one line to other line|Should render callout correctly on mouse move|Enzyme & RTL|
+|- Mouse over on a line with customized callout|Should render customized callout on mouseover|Enzyme|
+|- Customized callout on a line |Should render customized callout correctly on mouseover|Enzyme|
+|- Customized callout on a line from one line to other line|Should render customized callout for stack on mouseover|Enzyme|
+|Test 5: Render empty chart aria label div when chart is empty|||
+|- Vertical bar chart mounted with non-empty data|No empty chart aria label div rendered|Enzyme|
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:39]: This is line chart

Comment (AtishayMsft) [reply packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:39]: Fix in other places also

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+|- Specify line data|Should render line with the data provided|RTL|
+|- Hover mouse over the data points|Should highlight the data points (No callout is rendered when we hover only on the line. Callout appears on hover over the bars only.)|RTL|
+|Test 8: [Sub-Component]: Legends|||
+|- Hide legends |Should not show any rendered legends |RTL|
+|- Hover mouse over line legends|Should reduce the opacity of the other lines|RTL|
+|- Hover mouse over line legends|Should reduce the opacity of the other legends|RTL|
+|- Single mouse click on legends|Should select legend on single mouse click on respective legend|RTL|
+|- Double mouse click on legends|Should deselect legend on double mouse click on respective legend|RTL|
+|Test 9: [Sub-Component]: Callout|||
+|- Hover mouse over a bar|Should call the handler on mouse over bar|RTL|
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:54]: There is no bar in line chart

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+**Line Chart – Component test plan**
+
+**Sub-components: Line, x-axis, y-axis, Event, Time Range, Callout, Drop down and Legend**
+
+1. **Line: Line data, Line color (multi colors), Line label (show/hide)**
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:5]: Add tests for time range, event annotations, gaps in line and custom date axis locale.

Comment (v-baambati) [reply packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:5]: Sure i will add remaining tests

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md (hunk 2 above)
Comment (v-baambati) [reply packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md]: updated

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md (hunk 3 above)
Comment (v-baambati) [reply packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md]: Updated in all other places

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md (hunk 5 above)
Comment (v-baambati) [reply packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:5]: Added tests for Time range and gaps

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md (hunk 4 above)
Comment (v-baambati) [reply packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md]: updated description

File: packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+|                        - Rotate x-axis labels                         |                                  Should rotate the x-axis labels by 39 degrees                                  |      RTL      |
+|              Test 8: [Sub-Component]: Screen resolution               |                                                                                                                 |               |
+|              - Increase the screen resolution (zoom in)               |                                       Should remain unchanged on zoom in                                        |      RTL      |
+|              - Decrease the screen resolution (zoom out)              |                                       Should remain unchanged on zoom out                                       |      RTL      |
+|                 Test 10: Theme changed to Dark Theme                  |                                           Should reflect theme change                                           |      RTL      |
```
Comment (srmukher) [inline packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md:57]: Please review the Test case Numbers. For example: After test case 4, test case 6 is given, test case 9 is also absent.

**Review by srmukher — APPROVED** [review]

File: packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
+**Area Chart – Component test plan**
+
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md:2]: There is a conflict in AreaChart.base.tsx file. Can you resolve it.

File: packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+|                         Test 4: Mouse events                          |                                                                                                                 |               |
+|                         - Mouse over on Area                          |                                  Should render callout correctly on mouseover                                   |    Enzyme     |
+|               - Mouse move from one area to other area                |                                  Should render callout correctly on mouse move                                  |    Enzyme     |
+|            - Mouse over on a area with customized callout             |                                  Should render customized callout on mouseover                                  |    Enzyme     |
+|                     - Customized callout on area                      |                             Should render customized callout correctly on mouseover                             |    Enzyme     |
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md:40]: what is the difference between Mouse over on a area with customized callout and Customized callout on area

File: packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+1. **Area: Area data, Area color (multi colors), Area label (show/hide)**
+1. **Legends: show/hide legends, highlight the corresponding Area on legend hover**
+1. **Callout: Default/custom callout**
+1. **Labels: x-Axis labels, y-Axis labels**
+
+|                            **Test steps**                             |                                                 **Validation**                                                  | **Tool used** |
+| :-------------------------------------------------------------------: | :-------------------------------------------------------------------------------------------------------------: | :-----------: |
+|                      Test 1: [Snapshot testing]                       |                                                                                                                 |               |
+|           - With only data prop, numerical data on x-axis.            |                                          Renders Area chart correctly                                           |      RTL      |
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md:13]: Add a test case with string data on x axis. For this case the chart should not work.

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+1. **Legends: show/hide legends, highlight the corresponding line on legend hover**
+1. **Callout: Default/custom callout**
+1. **Labels: x-Axis labels**
+
+|                                      **Test steps**                                       |                                                             **Validation**                                                              | **Tool used** |
+| :---------------------------------------------------------------------------------------: | :-------------------------------------------------------------------------------------------------------------------------------------: | :-----------: |
+|                                Test 1: [Snapshot testing]                                 |                                                                                                                                         |               |
+|                     - With only data prop, numerical data on x-axis.                      |                                                      Renders line chart correctly                                                       |      RTL      |
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:13]: Add a test case with string data on x axis. The chart should not work in this case.

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+|                             - HideLegend prop set to “false”                              |                                              Should mount legend when hideLegend is false                                               |    Enzyme     |
+|                             - HideTooltip prop set to “true”                              |                                            Should not mount callout when hideTootip is true                                             |    Enzyme     |
+|                             - HideTooltip prop set to “false”                             |                                              Should mount callout when hideTootip is false                                              |    Enzyme     |
+|                        - onRenderCalloutPerStack prop is not given                        |                                                Should not render onRenderCalloutPerStack                                                |    Pending    |
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:26]: Are these pending cases fixed?

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+|                                   Test 4: Mouse events                                    |                                                                                                                                         |               |
+|                                  - Mouse over on a line                                   |                                              Should render callout correctly on mouseover                                               | Enzyme & RTL  |
+|                         - Mouse move from one line to other line                          |                                              Should render callout correctly on mouse move                                              | Enzyme & RTL  |
+|                      - Mouse over on a line with customized callout                       |                                              Should render customized callout on mouseover                                              |    Enzyme     |
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:35]: what is the difference between - Customized callout on a line and Mouse over on a line with customized callout

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+|          - prop changes: Mount line chart with empty data and then set the props          |                                             Render function should have been called 3 times                                             |    Enzyme     |
+|                               Test 7: [Sub-Component]: Line                               |                                                                                                                                         |               |
+|                                    - Specify line data                                    |                                                Should render line with the data provided                                                |      RTL      |
+|                            - Hover mouse over the data points                             | Should highlight the data points (No callout is rendered when we hover only on the line. Callout appears on hover over the lines only.) |      RTL      |
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:46]: What is this test case? We do show the callout when hovered over the line and not the datapoint.

File: packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
+**Line Chart – Component test plan**
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/LineChart/ComponentTests.md:1]: Add test case for - gaps in line, events on line and colorFillBars.
The usage is there in our demo site.

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx [hunk 15]
```diff
@@ -668,9 +668,8 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
               {...this._getClickHandler(this._points[i].onLineClick)}
               opacity={1}
               role="img"
-              aria-label={`${legendVal}, series ${i + 1} of ${this._points.length} with ${
-                this._points[i].data.length
-              } data points.`}
+              aria-label={`${legendVal}, series ${i + 1} of ${this._points.length} with ${this._points[i].data.length
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/LineChart/LineChart.base.tsx:671]: Why are there changes in this file

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx [hunk 16]
```diff
@@ -1171,15 +1171,15 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
...
+  private _handleHover(
     x: number | Date,
     y: number | Date,
     lineHeight: number,
     xAxisCalloutData: string,
     circleId: string,
     xAxisCalloutAccessibilityData: IAccessibilityProps,
     mouseEvent: React.MouseEvent<SVGElement>,
-  ) => {
+  ) {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/LineChart/LineChart.base.tsx:1182]: I am still seeing changes in this file.

File: packages/react-components/react-nav-preview/src/components/Nav/index.ts [hunk 17]
```diff
@@ -1,5 +0,0 @@
-export * from './Nav';
-export * from './Nav.types';
```
Comment (AtishayMsft) [inline packages/react-components/react-nav-preview/src/components/Nav/index.ts:2]: Remove changes from this line.

File: packages/react-charting/src/components/LineChart/LineChartRTL.test.tsx [hunk 18]
```diff
@@ -1,22 +1,626 @@
...
+      {
+        startX: new Date('2018/04/01'),
+        endX: new Date('2018/05/01'),
+      },
+    ],
+    applyPattern: true,
+  },
+];
+
+const pointsWithGaps: ILineChartPoints[] = [
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/LineChart/LineChartRTL.test.tsx:93]: where is the gaps object defined in pointsWithGaps

File: packages/react-components/react-nav-preview/src/components/Nav/index.ts (hunk 17 above)
Comment (AtishayMsft) [reply packages/react-components/react-nav-preview/src/components/Nav/index.ts:2]: The PR build is failing due to this change

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 19]
```diff
@@ -890,15 +890,15 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
     });
   };
 
-  private _getAriaLabel = (lineIndex: number, pointIndex: number): string => {
+  private _getAriaLabel(lineIndex: number, pointIndex: number): string {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:893]: I think this change is not needed.

File: packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md [hunk 20]
```diff
@@ -0,0 +1,61 @@
+**Area Chart – Component test plan**
+
+**Sub-components: Area, x-axis, y-axis and Legend**
+
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/AreaChart/ComponentTests.md:4]: Change file not present. PR build is failing

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx (hunk 19 above)
Comment (srmukher) [reply packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:893]: yes, this change is required as we have to mock the getAriaLabel function for Area chart and if we keep the function anonymous using the arrow function, jest cannot spy on that function.

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**